File: 파이썬/보코테/완전탐색/차이를 최대로/Main2.py
import sys


def next(arr):
    k = -1
    target = -1
    for i in range(len(arr) - 1):
        if arr[i] < arr[i + 1]:
            k = i

    # 내림차순 정렬로 되어 있는 경우
    if k == -1:
        return [-1]

    for i in range(k, len(arr)):
        if arr[k] < arr[i]:
            target = i

    arr[k], arr[target] = arr[target], arr[k]

    arr = arr[:k + 1] + sorted(arr[k + 1:])

    return arr


N = int(input())
arr = list(map(int, sys.stdin.readline().split()))
arr.sort()

max = -1
while True:
    arr = next(arr)
    if arr == [-1]:
        break
    sum = 0
    for i in range(N - 1):
        sum += abs(arr[i + 1] - arr[i])
    if sum > max:
        max = sum

print(max)

File: Programmers/src/바킹독/이분탐색/좌표압축/Main2.java
package 바킹독.이분탐색.좌표압축;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;


public class Main2 {
    int N;
    int[] arr;
    int[] filtered;
    public static void main(String[] args) throws IOException {
        new Main2().sol();
    }

    private void sol() throws IOException {
        setUp();

        StringBuilder sb = new StringBuilder();

        for (int target : arr) {
            sb.append(bs(target)+" ");
        }

        System.out.println(sb);
    }

    private int bs(int num) {
        int left = 0;
        int right = filtered.length;
        while (left < right) {
            int mid = (left + right) / 2;
            if (num <= filtered[mid]) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return left;
    }

    private void setUp() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new int[N];


        String[] input = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(input[i]);
        }

        filtered = Arrays.stream(arr).sorted().distinct().toArray();

    }
}
